controllers/news-controller.js

```diff
@@ -7,3 +7,17 @@ exports.getTopics = (req, res, next) => {
     })
 }
 
+exports.getArticleById = (req, res, next) => {
+    const {article_id} = req.params;
+    fetchArticleById(article_id).then((article) =>{
+        res.status(200).send({article})
+    }).catch((err) => {
+        if (err.status ===404){
+            res.status(404).send({ message: 'article does not exist'})
```

Comment:
this should be handled by a Promise.reject in your model. You shouldn't need this conditional logic :)